### PR TITLE
Add more examples to runs search docs page

### PR DESCRIPTION
**Pull request by apurva-koti**
Signed-off-by: Apurva Koti <apurva.koti@databricks.com>

## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Add more examples to the runs search docs page, to cover things like tag names with spaces, use of IN and LIKE, and other things we don't currently have examples for

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests (describe details, including test results, below)

image

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ]  No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Provided more examples of valid runs search queries in the associated documentation page.

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/search-runs.rst [hunk 1]
```diff
@@ -29,17 +29,41 @@ Example Expressions
 
     metrics.accuracy > 0.92
 
-- Search for runs created using a Logistic Regression model, a learning rate (lambda) of 0.001, and recorded error metric under 0.05.
+- Search for all failed runs.
```
[the remark this answers is not in the document]
Comment (apurva-koti) [reply docs/source/search-runs.rst:32]: Ya, although this was a pre-existing example. Should i change this or add that one on top?

File: docs/source/search-runs.rst [hunk 2]
```diff
@@ -29,17 +29,41 @@ Example Expressions
...
-    attributes.status = "FAILED"
+    metrics.`f1 score` > 0.5
+
+- Search for runs created by user 'john@mlflow.com'.
+
+  .. code-block:: sql
+
+    tags.`mlflow.user` = 'john@mlflow.com'
+
+- Search for runs whose ``model`` tag starts with ``sklearn``.
```
Comment (dbczumar) [inline docs/source/search-runs.rst:50]:
```suggestion
- Search for runs with models trained using scikit-learn (assumes the run author has set a tag called `model` whose value starts with `sklearn`).
```

File: docs/source/search-runs.rst [hunk 3]
```diff
@@ -29,17 +29,41 @@ Example Expressions
...
+
+  .. code-block:: sql
+
+    attributes.`run_name` IN ('alpha', 'beta', 'gamma')
+
+- Search for runs created using a Logistic Regression model, a learning rate (lambda) of 0.001, and recorded error metric under 0.05.
+
+  .. code-block:: sql
+
+    params.model = "LogisticRegression" and params.lambda = "0.001" and metrics.error <= 0.05
```
Comment (dbczumar) [inline docs/source/search-runs.rst:66]: We use model as a tag above, so it's a bit unclear what the best practice is. Can we remove `params.model`?

File: docs/source/search-runs.rst (hunk 2 above)
Comment (dbczumar) [inline docs/source/search-runs.rst:50]: Can we add an ILIKE example as well?

File: docs/source/search-runs.rst [hunk 4]
```diff
@@ -29,17 +29,41 @@ Example Expressions
```
Comment (dbczumar) [inline docs/source/search-runs.rst:29]: @apurva-koti Nice work! Can we also add examples for searching by attribute columns matching the row headers (i.e. relates to the support you're adding to match the column names properly)?

Comment (apurva-koti) [reply docs/source/search-runs.rst:29]: @dbczumar there will be a couple more changes to this page to support new features (including that), but I don't think we should merge them before those features exist

File: docs/source/search-runs.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (dbczumar) [reply docs/source/search-runs.rst:32]: Let's add that one on top :)

File: docs/source/search-runs.rst (hunk 4 above)
Comment (dbczumar) [reply docs/source/search-runs.rst:29]: @apurva-koti Got it! Can we file an internal JIRA to track updating these docs when we add the new capabilities?

Comment (apurva-koti) [reply docs/source/search-runs.rst:29]: DMed earlier :)

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @apurva-koti !

**Pull request merged**